### PR TITLE
Host address configurable

### PR DESCRIPTION
**Pull request by gbalke**
Hard-coding the host address was a bit messy. I'm thinking of turning this into a list and cameras will be allocated to the available addresses in a rolling fashion. This way we can have N number of connections and distribute the load.

### CODE REVIEW

**Comment by luca-della-vedova** [comment]
From what I understand it would actually be a list of server IPs right? To reduce future changes to the data structures and function definitions maybe we can already set it as a vector of length 1?

**Comment by gbalke** [comment]
I'm going to leave this PR here for sake of size. I think the next step is adjusting the Ethernet Client to spawn threads for each socket and then allocate sockets on a rolling basis to send off messages. This starts entering scheduler territory so I think we should start with the simplest solution and improve from there. My first idea is to just block until a socket is available to send the message. The second is to establish a queue. My main concern with the queue is that we should replace the previous request if it is not served by the time that the primary thread rolls around again. This is highly unlikely but still quite possible if bandwidth is less than the flood gates.

File: ovc5/software/camera_device_driver/include/ovc5_driver/ethernet_driver.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ class EthernetClient
   const char *cam_data_type = "rggb16";
 
 public:
-  EthernetClient(int port = 12345);
+  EthernetClient(std::vector<std::string> server_ips, int port = 12345);
```
Comment (luca-della-vedova) [inline ovc5/software/camera_device_driver/include/ovc5_driver/ethernet_driver.hpp:27]: Nit, `const&`

**Comment by luca-della-vedova** [comment]
> I'm going to leave this PR here for sake of size. I think the next step is adjusting the Ethernet Client to spawn threads for each socket and then allocate sockets on a rolling basis to send off messages. This starts entering scheduler territory so I think we should start with the simplest solution and improve from there. My first idea is to just block until a socket is available to send the message. The second is to establish a queue. My main concern with the queue is that we should replace the previous request if it is not served by the time that the primary thread rolls around again. This is highly unlikely but still quite possible if bandwidth is less than the flood gates.

Would this mean that a camera could send an image to a socket in one iteration and another one in the following? I guess the server would use the header to know what to do with them?
My original idea was estimating the bandwidth needed by any camera through the camera parameters (width * height * bytes_per_pixel * fps) and using it to calculate how busy every socket is and assign probed cameras to the least busy socket. However, if this works it sounds like it might make a more efficient use of bandwidth.
I think it could still be interesting in the future to do the bandwidth estimation though, just to make sure there is no saturation of the links and to either warn the user or throttle the imagers if there is (for example if, in an extreme case, the imagers run at 60 fps but we only have bandwidth to send 15 fps and we have 4 framebuffers, we might end up sending half a new and half an old frame, so it might be good to warn the user and throttle the imager).
Anyway apart from the derailing this looks good to me.

**Review by luca-della-vedova — APPROVED** [review]

**Pull request merged**